src/ocrmypdf/ocr_engine/rapidocr.py

```diff
@@ -0,0 +1,440 @@
+
+    @staticmethod
+    def get_orientation(input_file: Path, options) -> OrientationConfidence:
+        """Return the orientation detected by RapidOCR."""
+        # RapidOCR doesn't have built-in orientation detection like Tesseract
+        # Returning default value - could be enhanced with custom algo
+        return OrientationConfidence(angle=0, confidence=0.0)
+
+    @staticmethod
+    def get_deskew(input_file: Path, options) -> float:
+        """Return deskew angle (degrees)."""
+        # Return 0 as RapidOCR doesn't have direct deskew calculation
+        return 0.0
+
+    @staticmethod
+    def generate_hocr(
+        input_file: Path, output_hocr: Path, output_text: Path, options
+    ) -> None:
+        """Generate HOCR file from the OCR results."""
+        if not RAPIDOCR_AVAILABLE:
+            error_msg = f"rapidocr_onnxruntime is not installed: {IMPORT_ERROR_MESSAGE}"
+            log.error(error_msg)
+            raise ImportError(
+                f"{error_msg}\nInstall it with: pip install rapidocr_onnxruntime"
+            )
+
+        try:
+            # Load image
+            image = Image.open(input_file)
+            # Convert to numpy array for RapidOCR
+            img_array = np.array(image)
+
+            # Pre-process the image to improve detection
+            # Increase contrast
+            enhancer = ImageEnhance.Contrast(image)
+            image = enhancer.enhance(1.5)  # Increase contrast by 50%
+
+            # Or try padding the image
+            padded = Image.new(
+                image.mode, (image.width, image.height + 20), (255, 255, 255)
+            )
+            padded.paste(image, (0, 0))
+            img_array = np.array(padded)
+
+            # Initialize RapidOCR with configured options
+            rapidocr = RapidOCR(
+                use_angle_cls=getattr(options, 'rapidocr_use_angle_cls', False),
+                lang=getattr(options, 'rapidocr_lang', 'en'),
+            )
+
+            # Perform OCR - handle different return value patterns
+            log.debug(f"Running RapidOCR on image with shape: {img_array.shape}")
+            result = rapidocr(img_array)
+            log.info(f"RapidOCR returned result type: {type(result)}")
+
+            # Print complete result for debugging
+            if isinstance(result, list):
+                log.info(f"Got list with {len(result)} items")
+                if len(result) > 0:
+                    sample_item = result[0]
+                    log.info(
+                        f"First result item type: {type(sample_item)}, value: {sample_item}"
+                    )
+            elif isinstance(result, tuple):
+                log.info(f"Got tuple with {len(result)} items")
+                for i, item in enumerate(result):
+                    if i < 2:  # Only log first two items to avoid excessive output
+                        log.info(f"Tuple item {i}: type={type(item)}, value={item}")
```

Comment:
Is there a way to ask rapidocr to always return the same data type? 
If not, turn the tuple into a list with something like
`result = [result]` and eliminate all of the branching based on return value.